src/Controller/SpaceController.php

```diff
@@ -28,16 +28,38 @@ public function index(Request $request, SpaceRepository $spaceRepository, ?strin
     {
         $form = $this->createForm(SearchType::class, null, array('method' => 'GET'));
         $form->handleRequest($request);
-        $spaces = $spaceRepository->findAll();
 
         if ($form->isSubmitted() && $form->isValid()) {
             $location = $form->get('location')->getData();
-
-            $spaces = $spaceRepository->findByLocation($location);
+            return $this->redirectToRoute('space_search', array(
+                'location' => $location,
+            ));
```

Comment:
if you're not using location here, don't declare a form in index. if you're using location here, don't redirect to another route. you should take a look to embedded controllers